scheduling/allocator.rb

```diff
@@ -597,7 +610,15 @@ def map_volumes_to_devices
 
     def create_storage_volumes(vm, vm_host)
       @request.storage_volumes.each do |disk_index, volume|
-        spdk_installation_id = StorageAllocation.allocate_spdk_installation(vm_host.spdk_installations)
+        if vm_host.vhost_block_backends.sum(&:allocation_weight) == 0
```

Comment:
No need to sum at all, just check if there are no positive allocation weights (assuming that negative allocation weights are not supported):

```suggestion
        if vm_host.vhost_block_backends_dataset.exclude(allocation_weight: 0).empty?
```

Reply:
Done.